orgbee/src/app/login/login.component.html

```diff
@@ -39,64 +54,137 @@ <h4>
 </div>
 
 <!-- Signup Popup -->
-<div *ngIf="showSignup" class="popup">
-  <div class="popup-content">
-    <span class="close" (click)="hideSignupPopup()">&times;</span>
-    <h3>Sign Up</h3>
-    <div class="signup-form">
-      <form [formGroup]="signupForm" (ngSubmit)="onSignupSubmit()">
-        <div class="form-row">
-          <div>
-            <label for="signup-firstname">First Name</label>
-            <input type="text" formControlName="firstName" />
+@if(showSignup){
+  <div class="popup">
+    <div class="popup-content">
+      <span class="close" (click)="hideSignupPopup()">&times;</span>
+  
+      <h3>Sign Up</h3>
+  
+      <div class="signup-form">
+        <form [formGroup]="signupForm" (ngSubmit)="onSignupSubmit()">
+          <div class="form-row">
+            <div>
+              <label for="signup-firstname">First Name</label>
+              <input type="text" formControlName="firstName" />
+              @if(firstNameControl.invalid && firstNameControl.touched){
+                <div class="inc">
+                    @if(firstNameControl.errors?.['required']){
+                        <div>First name is required</div>
+                    }
+                </div>
+              }
+            </div>
+  
+            <div>
+              <label for="signup-lastname">Last Name</label>
+              <input type="text" formControlName="lastName" />
+              @if(lastNameControl.invalid && lastNameControl.touched){
+                <div class="inc">
+                    @if(lastNameControl.errors?.['required']){
+                        <div>Last name is required</div>
+                    }
+                </div>
+              }
+            </div>
           </div>
-          <div>
-            <label for="signup-lastname">Last Name</label>
-            <input type="text" formControlName="lastName" />
+  
+          <div class="form-row">
+            <div>
+              <label for="signup-studentnumber">Student Number</label>
+              <input type="text" formControlName="studentNumber" />
+              @if(studentNumberControl.invalid && studentNumberControl.touched){
+                <div class="inc">
+                    @if(studentNumberControl.errors?.['required']){
+                        <div>Student number is required</div>
+                    }
+                </div>
+              }
+            </div>
+  
+            <div>
+              <label for="signup-email">Email</label>
+              <input type="email" formControlName="email" />
+              @if(signInEmailControl.invalid && signInEmailControl.touched){
+                <div class="inc">
+                    @if(signInEmailControl.errors?.['required']){
+                        <div>Email is Required.</div>
+                    }
+                    @if(signInEmailControl.errors?.['email']){
+                        <div>Invalid Email format.</div>
+                    }
```

Comment:
signInEmailControl in sign up form, maybe there's a typo somewhere in your typescript, and if not then change it to signUpEmailControl